### PR TITLE
Adding support for app-services error reporting

### PR DESCRIPTION
**Pull request by bendk**
Added bridge code for a new app-services error reporting system that we're currently working on.

I'm hoping that we can use this to improve how our errors are grouped in Sentry, which should make them easier to manage.  It also adds breadcrumb support, which will hopefully allow us to address them faster as well.

See https://github.com/mozilla/application-services/pull/4979 for more info.

### Pull Request checklist

- [ ] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/android-components/blob/main/docs/changelog.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/main/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Milestone**: Make sure issues closed by this pull request are added to the [milestone](https://github.com/mozilla-mobile/android-components/milestones) of the version currently in development.
- **Breaking Changes**: If this is a breaking change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### CODE REVIEW

**Review by jonalmeida — COMMENTED** [review]
Overall this looks fine to me after having a first pass and a quick peek at the a-s change.

**Comment by bendk** [comment]
Just updated this one.  The a-s version was bumped earlier today so this PR doesn't need to do it.

**Comment by bendk** [comment]
Just updated this one.  The a-s version was bumped earlier today so this PR doesn't need to do it.

File: components/support/rusterrors/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+<!-- This Source Code Form is subject to the terms of the Mozilla Public
+   - License, v. 2.0. If a copy of the MPL was not distributed with this file,
+   - You can obtain one at http://mozilla.org/MPL/2.0/.  -->
+<manifest xmlns:android="http://schemas.android.com/apk/res/android"
+    package="mozilla.components.support.appserviceserrors" />
```
Comment (jonalmeida) [inline components/support/rusterrors/src/main/AndroidManifest.xml:5]: nit: keep the package name consistent with the structure.
```suggestion
    package="mozilla.components.support.rusterrors" />
```

File: docs/changelog.md [hunk 2]
```diff
@@ -54,6 +54,9 @@ permalink: /changelog/
 * **support-ktx**
   * 🌟️ Add support for optionally persisting the default value when `stringPreference` is used to read a string from SharedPreferences. [issue #12207](https://github.com/mozilla-mobile/android-components/issues/12207).
 
+* **support-rusterrors**
```
Comment (jonalmeida) [inline docs/changelog.md:57]: Let's bump this up to the latest release version 103 before landing.

**Review by jonalmeida — APPROVED** [review]
LGTM - let's just fix the package name in the AndroidManifest before merging.

**Pull request merged**